test/e2e/volume_provisioning.go

```diff
@@ -125,6 +126,56 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
 
 			testDynamicProvisioning(c, claim)
 		})
+
+		It("should test that deleting a claim before the volume is provisioned deletes the volume. [Volume]", func() {
+			// This case tests for the regressions of a bug fixed by PR #21268
+			// REGRESSION: Deleting the PVC before the PV is provisioned can result in PV
+			// not being deleted
+
+			// Given the small race window, attempt to produce it iteratively
+			var raceAttempts int = 100
+
+			By("creating a StorageClass")
+			class := newStorageClass()
+			class.Name = ns + "-sc" // pseudo-random Storage Class name
+			class, err := c.Storage().StorageClasses().Create(class)
+			defer c.Storage().StorageClasses().Delete(class.Name, nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("Creating and deleting PersistentVolumeClaims for %d iterations", raceAttempts))
+			for i := 0; i < raceAttempts; i++ {
+				claim := newClaim(ns, false)
```

Comment:
Good point.  I was thinking it would result in name collisions but that doesn't get assigned until the Create call.  Will tweak that.